### PR TITLE
Fix Test Cases: `test_http` for Py3.9+, `test_json_response_json` for ujson 5.4.0+, and `test_zero_downtime`; Test Case Type Annotations

### PR DESCRIPTION
**Pull request by ChihweiLHBird**
The test case was fixed.

### CODE REVIEW

File: tests/test_json_encoding.py [hunk 1]
```diff
@@ -3,11 +3,16 @@
 from dataclasses import asdict, dataclass
 from functools import partial
 from json import dumps as sdumps
+from typing import Dict
 
 import pytest
 
+from packaging import version
```
Comment (ahopkins) [inline tests/test_json_encoding.py:10]: No need to add a dependency. Just parse the version by hand. We do it in a couple of places.

**Review by ahopkins — CHANGES_REQUESTED** [review]

File: tests/test_json_encoding.py (hunk 1 above)
Comment (ChihweiLHBird) [reply tests/test_json_encoding.py:10]: `packaging` is used by setuptools so I thought it is everywhere. And sure, I can manually parse it.

File: tests/test_json_encoding.py [hunk 2]
```diff
@@ -75,7 +84,13 @@ def test_json_response_ujson(payload):
         json(payload)
 
 
-@pytest.mark.skipif(NO_UJSON is True, reason="ujson not installed")
+@pytest.mark.skipif(
```
Comment (ahopkins) [inline tests/test_json_encoding.py:87]: We are going need to rethink this test. But will merge it for now.

File: tests/test_unix_socket.py [hunk 3]
```diff
@@ -162,11 +166,11 @@ async def client(app, loop):
 app_multi = Sanic(name="test")
 
 
-def handler(request):
+def handler(request: Request):
     return text(f"{request.conn_info.server}")
 
 
-async def client(app, loop):
+async def client(app: Sanic, loop: AbstractEventLoop):
```
Comment (ahopkins) [inline tests/test_unix_socket.py:173]: There really is no reason to start annotating all of these. Especially those embedded handlers.

**Review by ahopkins — APPROVED** [review]

**Pull request merged**